src/Rules/Violations.php

```diff
@@ -108,13 +108,13 @@ public function toArray(): array
 
     public function remove(self $violations): void
     {
-        $this->violations = array_udiff(
+        $this->violations = array_values(array_udiff(
```

Comment:
is it possible to simplify this code for you?

Reply:
I could remove the `array_values` method. It is only added because `array_udiff` maintain the old array keys for the violations.

Reply:
I mean to refactor the function to be more explicit. Now we have your test and we can improve the code :) 
We can do it also in another PR.

Reply:
In that case, I prefer to not modify that in this PR, if a new version with this fix could be released.